src/audio/module_adapter/module_adapter.c

```diff
@@ -35,24 +35,15 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	int ret;
 	struct comp_dev *dev;
 	struct processing_module *mod;
+	struct ipc_config_process *ipc_module_adapter = spec;
 	unsigned char *data;
 	uint32_t size;
 
-	switch (config->type) {
```

Comment:
@ranj063 was type used to support legacy or was there an issue with UUID for all component/modules ?

Reply:
@dbaluta @lgirdwood I can use UUID as well but the list of cases in the switch/case block will expand rapidly as and when we start adding more and more processing modules. On the contrary, since we're forcing the use of module adapter for all processing components, its quite a simple change in topology to just set their type as MODULE_ADAPTER and capture them all in one case. Im fine either way, just want to know make sure you're OK with adding a whole bunch of UUID's here

Reply:
@dbaluta I have a PR pending to use the module adapter for the volume component, It will not work this change.

Reply:
@ranj063 IIUC using UUID should be a lookup and not a large expanding switch() i.e.
1) Each module registers it's UUID (today at build time)
2) We do a UUID lookup for each module topology load.

I agree, we should have a small switch() that rejects type != MODULE_ADAPTER